// noice/index.js
{
  function isIterable (arg) {
    return (arg[0] && arg.slice && !isNaN(arg.length))
  }

  function each (arr, proc, skip = 0) {
    for (let i = skip; i < arr.length; i++) {
      proc(arr[i], i, arr.length, arr)
    }
  }

  function random (min = 0, max = 1) {
    return Math.round((Math.random() * max) + min)
  }

  function range (...args) {
    let low, high
    if (args.length === 1) {
      low = 0
      high = args[0]
    } else {
      low = args[0]
      high = args[1]
    }
    let arrayLength = high - low
    let o = Array(arrayLength)
    for (let i = low, index = 0; i < high; i++, index++) {
      o[index] = i
    }
    return o.slice()
  }

  function pick (arr) {
    return arr[random(0, arr.length - 1)]
  }

  function replace (arg, search, repl = '') {
    if (Array.isArray(search) && Array.isArray(repl)) {
      var res = arg.slice()
      for (var i = 0; i < search.length; i++) {
        res = res.split(search[i]).join(repl[i])
      }
      return res
    }
    return arg.slice().split(search).join(repl)
  }

  function isDefined (arg) {
    return !(typeof arg === 'undefined')
  }
  module.exports = {
    isIterable,
    each,
    random,
    range,
    pick,
    replace,
    isDefined
  }
}
